Added encoded character section

osisoft-mbishop opened the pull request:
@dixeyja , I added the doc section for stream id sanitation related to https://dev.azure.com/osieng/engineering/_workitems/edit/249473. Please review and let me know if anything needs corrections.

dixeyja commented on content/configuration/pi-adapter-for-azure-event-hubs-data-selection-configuration.md:
```diff
@@ -61,10 +61,28 @@ The full schema definition for the Azure Event Hubs data selection configuration
 
+### Special characters encoding
+
+The adapter encodes special characters used in the **StreamId** parameter string before sending it to configured endpoints. The adapter substitutes the following encoded characters for special characters:
```
I'm not actually sure if this is correct. The changes I made are only sanitize the streamID like this when a Discovery is performed or when a data selection item is created without a streamID supplied by the user.

dixeyja replied:
This should actually specify that this only applies when the StreamId parameter is left blank. It's not something the user has to consider while creating a StreamID, but it helps to understand why the default StreamID looks the way it does when they view their data selections.

(The following doesn't have to be documented, just an explanation of why this is the way it is)
We do a sort of encoding on the framework side which converts the same characters we do for the default Stream ID does into URL friendly characters, but does not exchange them for other characters entirely like we do for the default StreamIds. We do this default StreamId character exchange to avoid having to later do the URL character changes.

osisoft-mbishop commented:
Updates:

* Added a note to [Data Selection](https://osisoft-dev.zoominsoftware.io/bundle/pi-adapter-azure-event-hubs-added-encoded-character-section/page/configuration/pi-adapter-for-azure-event-hubs-data-selection-configuration.html) **StreamId** entry that the adapter encodes any special characters you add, points to [Special characters encoding](https://osisoft-dev.zoominsoftware.io/bundle/pi-adapter-azure-event-hubs-added-encoded-character-section/page/main/shared-content/configuration/egress-endpoints.html#special-characters-encoding).

* Added a note to  [Azure Event Hubs data source discovery] that for `streamID` query results, the text parser replaces special characters with a different character. Points to [Special character support](https://osisoft-dev.zoominsoftware.io/bundle/pi-adapter-azure-event-hubs-added-encoded-character-section/page/main/shared-content/configuration/text-parser/text-parser.html#special-characters-support).

Just a heads up, the updates to "Special character encoding" and "Special characters support" will not show in this pull request because the changes were made in a different repo.
